runtime/ftplugin/javascript.vim

```diff
@@ -26,13 +26,70 @@ setlocal comments=sO:*\ -,mO:*\ \ ,exO:*/,s1:/*,mb:*,ex:*/,://
 
 setlocal commentstring=//%s
 
-" Change the :browse e filter to primarily show Java-related files.
-if has("gui_win32")
-    let  b:browsefilter="Javascript Files (*.js)\t*.js\n" .
-		\	"All Files (*.*)\t*.*\n"
+" Change the :browse e filter to primarily show JavaScript-related files.
+if (has("gui_win32") || has("gui_gtk")) && !exists("b:browsefilter")
+    let b:browsefilter =
+                \ "JavaScript Files (*.js)\t*.js\n"
+                \ .. "JSX Files (*.jsx)\t*.jsx\n"
+                \ .. "JavaScript Modules (*.es, *.es6, *.cjs, *.mjs, *.jsm)\t*.es;*.es6;*.cjs;*.mjs;*.jsm\n"
+                \ .. "Vue Templates (*.vue)\t*.vue\n"
+                \ .. "JSON Files (*.json)\t*.json\n"
+                \ .. "All Files (*.*)\t*.*\n"
 endif
-       
-let b:undo_ftplugin = "setl fo< ofu< com< cms<" 
+
+" The following suffixes should be implied when resolving filenames
+setlocal suffixesadd+=.js,.jsx,.es,.es6,.cjs,.mjs,.jsm,.vue,.json
+
+" The following suffixes should have low priority
+"   .snap    jest snapshot
+setlocal suffixes+=.snap
+
+" Prepend node_modules/.bin to $PATH if applicable
+" Allows calling executables installed locally via npm/yarn
+let s:bin_dir = finddir('node_modules/.bin', '.;')
+if !empty(s:bin_dir)
+    let s:bin_dir = fnamemodify(s:bin_dir, ':p')
+
+    if $PATH !~ s:bin_dir
+        let $PATH = s:bin_dir .. ':' .. $PATH
```

Comment:
I think this does not work on Windows since path separator on Windows is `;`.

Reply:
You are right.

Reply:
Fixed. Do you know by any chance if there are other separators I should consider?

Reply:
There is no new commit in this branch (you might forget pushing the fix).

> Do you know by any chance if there are other separators I should consider?

AFAIK considering `:` and `;` is sufficient.

Reply:
There.